src/transformers/models/ernie_m/tokenization_ernie_m.py

```diff
@@ -0,0 +1,366 @@
+# coding=utf-8
+# Copyright 2023 Xuan Ouyang, Shuohuan Wang, Chao Pang, Yu Sun, Hao Tian, Hua Wu, Haifeng Wang and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tokenization classes for Ernie-M."""
+
+# Some of the code is Copied from original paddlenlp repository.(https://github.com/PaddlePaddle/PaddleNLP/blob/develop/paddlenlp/transformers/ernie_m/tokenizer.py)
+# and Albert directory.(https://github.com/huggingface/transformers/blob/main/src/transformers/models/albert/tokenization_albert.py)
+import io
+import unicodedata
+from typing import List, Optional
+
+import sentencepiece as spm
```

Comment:
As the tokenizer relies on `sentence_piece` the test  `is_sentence_piece_available` might be needed in the __init__ (instead of `is_tokenizer_available`.